### PR TITLE
Refactored splitter tests

### PR DESCRIPTION
**Pull request by tovbinm**
**Related issues**
Test code was too repetitive and was not testing splitter summary correctly.

**Describe the proposed solution**
- Added `SplitterSummaryAsserts` for asserting `SplitterSummary` classes 
- Refactored tests for `DataCutter`, `DataSplitter`, `DataBalancer` and `OpValidator` to use `SplitterSummaryAsserts`
- Minor idents and cleanups

**Describe alternatives you've considered**
NA

**Additional context**
TBD - also needs checks & tests for empty datasets - https://github.com/salesforce/TransmogrifAI/issues/161

### CODE REVIEW

File: core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala [hunk 1]
```diff
@@ -73,46 +73,51 @@ class OpValidatorTest extends FlatSpec with TestSparkContext {
...
-  val balancer = Option(new DataBalancer())
-  val cutter = Option(new DataCutter())
-
+  val cvStratifyCondition = cv.isClassification && cv.stratify
+  val tsStratifyCondition = ts.isClassification && ts.stratify
 
   Spec[OpCrossValidation[_, _]] should "stratify binary class data" in {
-    val splits = cv.createTrainValidationSplits(condition, binaryDS, label.name, balancer)
+    val balancer = new DataBalancer()
+    val splits = cv.createTrainValidationSplits(cvStratifyCondition, binaryDS, label.name, Some(balancer))
```
Comment (marcovivero) [inline core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala]: Minor comment: use `Option`

File: core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala [hunk 2]
```diff
@@ -73,46 +73,51 @@ class OpValidatorTest extends FlatSpec with TestSparkContext {
...
     }
-    balancer.get.summary.get.toMetadata() should not be new MetadataBuilder().build()
+    assertDataBalancerSummary(balancer.summary) { s =>
+      Some(s) shouldBe new DataBalancer().prepare(binaryDS).summary
+    }
   }
 
   it should "stratify multi class data" in {
-    val splits = ts.createTrainValidationSplits(condition, multiDS, multiLabel.name, cutter)
+    val splits = ts.createTrainValidationSplits(tsStratifyCondition, multiDS, multiLabel.name, Some(new DataCutter()))
```
Comment (marcovivero) [inline core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala]: Same

File: core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala [hunk 3]
```diff
@@ -73,46 +73,51 @@ class OpValidatorTest extends FlatSpec with TestSparkContext {
...
       assertFractions(multiClassProbabilities, validate)
     }
+    assertDataCutterSummary(new DataCutter().prepare(multiDS).summary)(_ => succeed)
   }
 
-
   Spec[OpTrainValidationSplit[_, _]] should "stratify binary class data" in {
-    val splits = ts.createTrainValidationSplits(condition, binaryDS, label.name, balancer)
+    val balancer = new DataBalancer()
+    val splits = ts.createTrainValidationSplits(tsStratifyCondition, binaryDS, label.name, Some(balancer))
```
Comment (marcovivero) [inline core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala]: Same

File: core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala [hunk 4]
```diff
@@ -73,46 +73,51 @@ class OpValidatorTest extends FlatSpec with TestSparkContext {
...
     }
-    balancer.get.summary.get.toMetadata() should not be new MetadataBuilder().build()
+    assertDataBalancerSummary(balancer.summary) { s =>
+      Some(s) shouldBe new DataBalancer().prepare(binaryDS).summary
+    }
   }
 
   it should "stratify multi class data" in {
-    val splits = cv.createTrainValidationSplits(condition, multiDS, multiLabel.name, cutter)
+    val splits = cv.createTrainValidationSplits(cvStratifyCondition, multiDS, multiLabel.name, Some(new DataCutter()))
```
Comment (marcovivero) [inline core/src/test/scala/com/salesforce/op/stages/impl/tuning/OpValidatorTest.scala]: Same

File: core/src/main/scala/com/salesforce/op/stages/impl/tuning/OpValidator.scala [hunk 5]
```diff
@@ -185,19 +184,21 @@ private[op] trait OpValidator[M <: Model[_], E <: Estimator[_]] extends Serializ
...
+   * @return Array[(Train, Test)]
    */
-  private[op] def createTrainValidationSplits[T](stratifyCondition: Boolean,
-    dataset: Dataset[T], label: String, splitter: Option[Splitter] = None): Array[(RDD[Row], RDD[Row])]
-
+  private[op] def createTrainValidationSplits[T](
+    stratifyCondition: Boolean,
+    dataset: Dataset[T],
+    label: String,
+    splitter: Option[Splitter]
```
Comment (kinfaikan) [inline core/src/main/scala/com/salesforce/op/stages/impl/tuning/OpValidator.scala]: No default?

Comment (tovbinm) [reply core/src/main/scala/com/salesforce/op/stages/impl/tuning/OpValidator.scala]: nope, it's a private abstract method,
